### PR TITLE
Update LegacyYarn support

### LINKED ISSUE #10: Use Legacy Fabric meta instead of hardcoding supported versions
https://meta.legacyfabric.net/

### PR DESCRIPTION
**Pull request by thecatcore**
Closes #10

### CODE REVIEW

File: src/main/kotlin/me/shedaniel/linkie/namespaces/LegacyYarnNamespace.kt [hunk 1]
```diff
@@ -1,96 +1,57 @@
...
-        workingLegacyFabricVersions.clear()
-        workingLegacyFabricVersions.addAll(legacyFabricVersions.asSequence()
-            .filter { it.value != null }
-            .map { it.key }
-        )
-        latestLegacyFabricVersions = workingLegacyFabricVersions.maxByOrNull { it.toVersion() }!!
+        val buildMap = LinkedHashMap<String, MutableList<YarnNamespace.YarnBuild>>()
+        json.decodeFromString(ListSerializer(YarnNamespace.YarnBuild.serializer()), URL("https://meta.legacyfabric.net/v2/versions/yarn").readText())
+                .forEach { buildMap.getOrPut(it.gameVersion) { mutableListOf() }.add(it) }
+        buildMap.forEach { (version, builds) -> builds.maxByOrNull { it.build }?.apply { legacyYarnBuilds[version] = this } }
```
Comment (shedaniel) [inline src/main/kotlin/me/shedaniel/linkie/namespaces/LegacyYarnNamespace.kt:55]: Reset legacyYarnBuilds first before this line?

File: src/main/kotlin/me/shedaniel/linkie/namespaces/LegacyYarnNamespace.kt [hunk 2]
```diff
@@ -1,96 +1,57 @@
...
                     }
                 }
             }
         }
     }
 
-    override fun getAllVersions(): Sequence<String> = workingLegacyFabricVersions.asSequence() + singleSequenceOf("1.2.5")
-    override fun getDefaultLoadedVersions(): List<String> = listOf()
+    override fun getAllVersions(): Sequence<String> = legacyYarnBuilds.keys.asSequence()
+    override fun getDefaultLoadedVersions(): List<String> {
```
Comment (shedaniel) [inline src/main/kotlin/me/shedaniel/linkie/namespaces/LegacyYarnNamespace.kt:42]: Please don't load any legacy yarn versions by default

**Review by shedaniel — CHANGES_REQUESTED** [review]

File: src/main/kotlin/me/shedaniel/linkie/namespaces/LegacyYarnNamespace.kt (hunk 1 above)
Comment (thecatcore) [reply src/main/kotlin/me/shedaniel/linkie/namespaces/LegacyYarnNamespace.kt:55]: Somehow it's not done in any Namespace...

**Review by shedaniel — APPROVED** [review]
LGTM, Thanks!

**Pull request merged**